commit 4243c2b29a4270842a19801ee6f9a252049aa380 (refs/changes/50/750/2)
Author: Trent Mick <trentm@gmail.com>
Date:   2016-10-20T12:04:36-07:00 (3 years ago)
    
    IMGAPI-597 imgapi crash when a docker v2 layer download fails
    Reviewed by: Todd Whiteman <todd.whiteman@joyent.com>
    Approved by: Todd Whiteman <todd.whiteman@joyent.com>

diff --git a/lib/images.js b/lib/images.js
index 179f24e..9b5e4e8 100644
--- a/lib/images.js
+++ b/lib/images.js
@@ -2681,22 +2681,6 @@ function _dockerDownloadAndImportImage(opts, callback) {
             var shasum = crypto.createHash('sha1');
             var md5sum = crypto.createHash('md5');
 
-            stream.on('end', function () {
-                req.log.debug('fs layer completed for %s', imgId);
-                ctx.resMessage({
-                    type: 'progress',
-                    payload: {
-                        id: imgId.substr(0, 12),
-                        status: 'Downloading',
-                        progressDetail: {
-                            current: size,
-                            total: size,
-                            start: startTs
-                        }
-                    }
-                });
-            });
-
             stream.on('data', function (chunk) {
                 if (ctx.downloadsCanceled) {
                     stream.destroy();
